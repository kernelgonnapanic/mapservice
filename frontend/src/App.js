import React from 'react'
import Navigation from './components/Navigation'
import AddPlace from './components/AddPlace'
import { store } from './components/redux/reducers'
import { Provider } from 'react-redux'

const App = () => {
	return (
		<Provider store={store}>
			<Navigation />
		</Provider>
	)
}

export default App
