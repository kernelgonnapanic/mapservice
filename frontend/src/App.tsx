import React from 'react'
import { Provider } from 'react-redux'
import { createGlobalStyle, ThemeProvider } from 'styled-components'
import Navigation from './components/Navigation/Navigation'
import { store } from './components/redux/reducers'


const theme = {
	colors: {
		mainDark: '#203A55',
		mediumGray: '#546577',
		white: '#FFFFFF',
		purple: 'rgb(209, 196, 233)',
		gray: 'hsl(220, 7%, 83%)',
	},
	headerHeight: '64px',
	barHeight: '75px',
	boxshadow: '0 3px 0 hsl(220, 7%, 83%) ',
	borderRadius: {
		small: 5,
		medium: "20px"
	},
	fontSize: {
		s: "14px",
		l: "16px",
		xl: "20px",
		xxl: "30px"
	}
}

const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Varela+Round&display=swap');


  html, body, #root {
 	 height: 100%
	}

  body {
	margin: 0;
	padding: 0;
	background-color: #FBFBFB;
	font-family: 'Varela Round', sans-serif;
  }

  *, *::before, *::after {
	  box-sizing: border-box
  }
`

const App: React.FC = () => {

	return (
		<Provider store={store}>
			<GlobalStyle />
			<ThemeProvider theme={theme}>
				<Navigation />
			</ThemeProvider>
		</Provider>
	)
}

export default App
