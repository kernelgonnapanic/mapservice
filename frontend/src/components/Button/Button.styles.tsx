import styled from 'styled-components'

export const Button = styled.button<{ backgroundColor: string }>`
	cursor: pointer;
	background-color: #ff4081;
	color: #fff;
	-webkit-box-shadow: none;
	-moz-box-shadow: none;
	box-shadow: none;
	text-transform: none;
	font-weight: 600;
	line-height: 42px;
	font-size: ${(props) => props.theme.fontSize.s};
	background-color: ${(props) => props.backgroundColor};
	height: 45px;
	padding: 0 22px;
	border: none;
	border-radius: ${(props) => props.theme.borderRadius.medium};
`
