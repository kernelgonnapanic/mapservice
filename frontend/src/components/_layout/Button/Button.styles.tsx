import styled from 'styled-components'

export const Button = styled.button`
	background-color: gray;
	max-width: 250px;
	max-height: 50px;
	color: white;
	font-weight: bold;
	border: none;
	cursor: pointer;
`
