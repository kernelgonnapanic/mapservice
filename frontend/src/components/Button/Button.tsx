import React from 'react'
import * as S from './Button.styles'

interface Props {
	text?: string
	onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void
	type?: 'button' | 'submit' | 'reset' | undefined
	color?: string
}

const Button: React.FC<Props> = ({
	text = 'WyÅ›lij',
	onClick = () => {},
	type = 'button',
	color = '',
}) => {
	return (
		<S.Button
			onClick={onClick}
			type={type}
			backgroundColor={color}
		>

			{text}
		</S.Button>
	)
}

export default Button
