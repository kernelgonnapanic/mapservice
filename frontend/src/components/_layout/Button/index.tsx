import React from 'react'
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'

const useStyles = makeStyles((theme: Theme) =>
	createStyles({
		button: {
			margin: theme.spacing(1),
		},
		input: {
			display: 'none',
		},
	}),
)

const MButton: React.FunctionComponent<{}> = () => {
	const classes = useStyles()

	return (
		<Button variant="contained" color="primary" className={classes.button}>
			Primary
		</Button>
	)
}

export default MButton
