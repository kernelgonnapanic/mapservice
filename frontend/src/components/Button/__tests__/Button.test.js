import React from 'react'
import ReactDOM from 'react-dom'
import {render, cleanup} from '@testing-library/react'
import Button from '../Button'
import {theme} from "../../../App";
import {ThemeProvider} from "styled-components";

afterEach(cleanup);

test("render Button component", async () => {
    const {asFragment} = render(<ThemeProvider theme={theme}><Button text="Hello!"/></ThemeProvider>)
    expect(asFragment()).toMatchSnapshot()
})

test("insert text into the button" , () =>{
    const container = document.createElement('div')
    ReactDOM.render(<ThemeProvider theme={theme}><Button text="testText"/></ThemeProvider>, container)
    expect(container.textContent).toMatch('testText')
})