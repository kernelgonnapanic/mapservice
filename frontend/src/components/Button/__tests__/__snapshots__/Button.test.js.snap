// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Button component 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm eqawEJ"
    type="submit"
  >
    Hello!
  </button>
</DocumentFragment>
`;
